### PR TITLE
feat: pull edit handler

### LINKED ISSUE #55: Verify that the user gets assigned on PR open
@Keyrxng rfc

What I think happened is:
- no user was assigned
- a pull-request was linked to this issue, which triggers an assignment of the user opening that PR
- the user somehow didn't get assigned (you can see from the timeline only the start message was triggered)
- `/start` assigned the user and displayed the message again

It should have just said "You are already assigned to this task" because it should have been the case.

_Originally posted by @gentlementlegen in https://github.com/ubiquibot/assistive-pricing/issues/32#issuecomment-2362687081_

### PR DESCRIPTION
**Pull request by gentlementlegen**
Resolves #55

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "Start | Stop",
   "description": "Assign or un-assign yourself from an issue.",
-  "ubiquity:listeners": ["issue_comment.created", "issues.assigned", "issues.unassigned", "pull_request.opened"],
+  "ubiquity:listeners": ["issue_comment.created", "issues.assigned", "issues.unassigned", "pull_request.opened", "pull_request.edited"],
```
Comment (0x4007) [inline manifest.json:4]: This is for adding `resolves #xxx` ?

**Review by 0x4007 — COMMENTED** [review]
Don't understand this change

**Comment by 0x4007** [comment]
Your update commentary needs to be under the pull not the issue 

> They all worked fine. The only problem I noticed and when you open a PR, then change the description and link an issue, the bot won't assign and post the comment, which believe should be fixed (dunno if v1 handled that).

So you're saying now the bot reads the issue number from the closing keyword and automatically assigns? I don't understand what the point of adding pull body edit is

**Comment by gentlementlegen** [comment]
It fixes the scenario where you open a PR (draft or not) with an empty description, and then you edit it adding the `Resolves #xx`) . Within github, it will link your PR as closing of the issue, but if we do not observe the event on the plugin, the kernel won't assign the user to the issue.

**Comment by 0x4007** [comment]
It should only be on the linked event. Right? It will "link" when they fix the pull body

**Comment by gentlementlegen** [comment]
You can try it, open a pull-request with an empty body, then modify it to link it to a PR, GitHub will properly create a link between the pull-request and the issue. There is no linked event afaik, so checking the pull-request modified allows us to check for the linked issue, if any.

**Review by 0x4007 — APPROVED** [review]

**Comment by gentlementlegen** [comment]
@0x4007 Seems that Jest is not ok with the changes? Was it broken before you closed the PR? I'll fix this.

**Comment by Keyrxng** [comment]
I just QA'd development and it is possible to have more than one linked issue in the PR and in my case it assigned me to both tasks and I thought of legit pieces of text that may use any of the below prefixes. I think we should handle this case but I can't think of a guaranteed way to do it.

image

image

I think the best way to do it maybe is to use the reference which appears first in the PR body? Otherwise if all the conditions of `/start` are valid then the user would be assigned to all the tasks linked which are.

**Comment by 0x4007** [comment]
Can you link your qa I don't understand your explanation.

**Comment by gentlementlegen** [comment]
@Keyrxng why not using GraphQl? It handles multiple linked pull-requests.

**Comment by Keyrxng** [comment]
- https://github.com/ubq-testing/command-start-stop/pull/23

I first comment with only "Resolves ..." and then I edit it to add more context regarding a related issue to which I get [assigned](https://github.com/ubq-testing/command-start-stop/issues/22). Because the conditions of `/start` were valid I was assigned the task while my intention may not have been that. 

It makes it so that you must be aware of not using these prefixes otherwise you might end up assigned.

> why not using GraphQl? It handles multiple linked pull-requests.

This is using the `development` branch which is using GraphQL. I'm not sure how you'd be able to accurately deduce from the GraphQL issue response objects which referenced task should be assigned as both might exist in the same repo etc. Unless you can tell from the GraphQL fetch which appears first ofc.

**Comment by 0x4007** [comment]
> It makes it so that you must be aware of not using these prefixes otherwise you might end up assigned.

Then don't use the prefixes? Or just unassign yourself if its a problem?

**Comment by Keyrxng** [comment]
> Then don't use the closing keywords? Or just unassign yourself if its a problem?

My point was that contributors partners etc will all need to be aware of these little gotchas or it's going to happen to others out in the wild, obviously I know to avoid it so that it doesn't happen to me.

But if it's a non-issue then no worries just thought I'd shine a light on it

**Comment by 0x4007** [comment]
If it follows GitHub behavior it's fine

I think if the pull body starts with a closing keyword then GitHub will link it

I'm not sure about in the middle but even so it's so specific that it's a non issue. It's not gonna happen on accident. 

Just test and match GitHub behavior
